service-clients-core/src/test/java/com/flipkart/poseidon/serviceclients/ServiceResponseDecoderTest.java

```diff
@@ -235,35 +233,13 @@ public void testDecodeHttpResponseDefaultResponse() throws Exception {
         when(StringUtils.convertStreamToString(stream)).thenReturn("error");
         exceptions.put("default", ServiceClientException.class);
 
-        ServiceResponse response = decoder.decode(mockHttpResponse);
-        assertThat(response.getException(), instanceOf(ServiceClientException.class));
-        Mockito.verify(mockLogger).warn("Non 200 response statusCode:{} response: {}","500", "error");
-
-    }
-
-    /**
-     *  Service returned un known status line(not in expected statusLines) and decode fails
-     * @throws Exception
-     */
-    @Test
-    public void testDecodeHttpResponseDefaultResponseDecodeFail() throws Exception {
-        HttpResponse mockHttpResponse = mock(HttpResponse.class);
-        StatusLine mockStatusLine = mock(StatusLine.class);
-        HttpEntity mockEntity = mock(HttpEntity.class);
-        InputStream stream = mock(InputStream.class);
-
-        when(mockStatusLine.getStatusCode()).thenReturn(500);
-        when(mockHttpResponse.getStatusLine()).thenReturn(mockStatusLine);
-        when(mockHttpResponse.getEntity()).thenReturn(mockEntity);
-        when(mockEntity.getContent()).thenReturn(stream);
-        mockStatic(StringUtils.class);
-        when(StringUtils.convertStreamToString(stream)).thenThrow(IOException.class);
-        exceptions.put("default", ServiceClientException.class);
-
-        exception.expect(IOException.class);
-        exception.expectMessage(equalTo("Non 200 response de-serialization error"));
-        ServiceResponse response = decoder.decode(mockHttpResponse);
-        Mockito.verify(mockLogger).error("Error de-serializing non 200 response");
```

Comment:
This test is not needed and was throwing an exception within mockito framework and still succeeding. StringUtils. convertStreamToString never throws IOException. That's why the global try-catch is also removed in ServiceResponseDecoder